examples/memory_basic_usage.py

```diff
@@ -0,0 +1,138 @@
+    """Simulate an expensive computation"""
+    time.sleep(5)
+    return data[column_index]
+
+
+expensive_computation_cached = memory.cache(expensive_computation_cached)
+tic = time.time()
+data_trans = expensive_computation_cached(data)
+toc = time.time()
+
+print('\nOur function took {:.2f} s to compute.'.format(toc - tic))
+print('\nThe transformed data are:\n {}'.format(data_trans))
+
+###############################################################################
+# At the first call, the results will be cached. Therefore, the computation
+# time correspond to the time to compute the results plus the time to dump the
+# results into the disk.
+
+tic = time.time()
+data_trans = expensive_computation_cached(data)
+toc = time.time()
+
+print('\nOur function took {:.2f} s to compute.'.format(toc - tic))
+print('\nThe transformed data are:\n {}'.format(data_trans))
+
+###############################################################################
+# At the second call, the computation time is largely reduced since that the
+# results are obtained by loading the data previously dumped on the disk
+# instead of recomputing the results.
+
+###############################################################################
+# Using :class:`joblib.Memory` with a method
+###############################################################################
+# 
+# :class:`joblib.Memory` is designed to work with pure functions. When you want
+# to cache a method within a class, you need to create and cache a pure
+# function and use it inside the class.
+
+
+def _expensive_computation_cached(data, column):
+    time.sleep(5)
+    return data[column]
+
+
+class Algorithm(object):
+    """A class which is using our pure function."""
+
+    def __init__(self, column=0):
+        self.column = column
+
+    def transform(self, data):
+        expensive_computation = memory.cache(_expensive_computation_cached)
+        return expensive_computation(data, self.column)
+
+
+transformer = Algorithm()
+tic = time.time()
+data_trans = transformer.transform(data)
+toc = time.time()
+
+print('\nOur function took {:.2f} s to compute.'.format(toc - tic))
+print('\nThe transformed data are:\n {}'.format(data_trans))
+
+###############################################################################
+
+tic = time.time()
+data_trans = transformer.transform(data)
+toc = time.time()
+
+print('\nOur function took {:.2f} s to compute.'.format(toc - tic))
+print('\nThe transformed data are:\n {}'.format(data_trans))
+
+###############################################################################
+# As expected, the second call to the ``transform`` method load the results
+# which have been cached.
+
+###############################################################################
+# Clean up cache directory
+###############################################################################
+
+import shutil
+try:
+    shutil.rmtree(cachedir)
```

Comment:
You could call `mem.clear()` I think

Reply:
what is this function?

Reply:
It's in the public API of memory. So you can use memory.clear() to flush all cache.

Reply:
I mean instead of the while import/try/except

Reply:
interesting. I was not aware of it. Thanks. I should read the doc before to write it :)